tests/coords/transform_coords_test.py

```diff
@@ -473,44 +600,73 @@ def make_binned():
     return binned
 
 
-def check_binned(da, original):
-    y = original.coords['y']
-    assert 'xy' not in original.bins.coords  # Buffer was copied
-    assert 'x' in original.bins.coords  # Buffer was copied for consume
-    assert sc.identical(da.bins.coords['xy'],
-                        (y * original.bins.coords['x']).rename_dims({'x': 'x2'}))
-    assert 'yy' not in da.bins.coords
-    assert sc.identical(da.coords['yy'], y * y)
+def test_binned_does_not_modify_inputs(binned_in_a_b):
+    _ = binned_in_a_b.transform_coords(['b2'], graph={'b2': 'b'})
+    assert 'b' in binned_in_a_b.coords
+    assert 'b' in binned_in_a_b.bins.coords
+    assert 'b2' not in binned_in_a_b.meta
+    assert 'b2' not in binned_in_a_b.bins.meta
 
 
-def test_binned():
-    binned = make_binned()
+def test_binned_with_slice_does_not_modify_inputs(binned_in_a_b):
+    # If input is sliced, transform_coords has to copy the buffer
+    original = binned_in_a_b['a', 0:1]
```

Comment:
Is it worth testing also with a non-range slice?

Reply:
It helps to ensure that there is no set of inputs where we accidentally change the input. So, yes...?

Should be easy to do